src/index.js

```diff
@@ -586,6 +624,35 @@ const toClusterInfo = ({
  */
 const getName = (file) => file.name
 
+/**
+ * @param {ReadableStream<Uint8Array>} stream
+ */
+const ndjsonParse = async function* (stream) {
+  const reader = stream.getReader()
+  const matcher = /\r?\n/
+  const decoder = new TextDecoder('utf8')
+  let buffer = ''
+  try {
+    while (true) {
+      const result = await reader.read()
+
+      if (result.done) {
+        break
+      }
+
+      buffer += decoder.decode(result.value, { stream: true })
+      const parts = buffer.split(matcher)
+      buffer = parts.pop() || ''
+      for (const part of parts) yield JSON.parse(part)
+    }
+  } finally {
+    reader.cancel()
+    reader.releaseLock()
+  }
+  buffer += decoder.decode()
```

Comment:
I'm new here, whey do we call `decoder.decode()` with no params here? Should the `stream` param be passed in as `false` as per 

> `stream` A boolean flag indicating that additional data will follow in subsequent calls to decode(). Set to true if processing the data in chunks, and false for the final chunk or if the data is not chunked. It defaults to false.
– https://developer.mozilla.org/en-US/docs/Web/API/TextDecoder/decode#parameters

maybe it doesn't matter in practice as we're not passing in any more data?

Reply:
It defaults to false, as it says there...

We could explicitly pass it I guess.

Reply:
ah no, i was skimming. I was trying to figure out why the call to `decoder.decode()` with no params but now i realise that we are implicitly signalling to decoder that we're done streaming. As I right this, it feels like yes, it would be clearer for the new reader if it was explicit.

Reply:
Yes correct - I can change to pass it if it makes understanding it easier.